lib/scripts.js

```diff
@@ -80,7 +80,9 @@ var scripts = {
     var script = [
       'local numericJobId = redis.call("INCR", KEYS[5])',
       'local jobId = ARGV[2] == "" and numericJobId or ARGV[2]',
-      'redis.call("HMSET", ARGV[1] .. jobId' + argvs.join('') + ')',
+      'local fullJobId = ARGV[1] .. jobId',
+      'if redis.call("EXISTS", fullJobId) == 1 then return jobId end',
```

Comment:
you could rename fullJobId to jobIdKey since it represents more accurately why we are doing the concatenation with ARGV[1] i